### ISSUE TITLE
The `AgentPool` resource looses the `agent_pool_id` and `token_value` after a refresh

### ISSUE THREAD

**Issue by aureq** [pulumi/pulumi-pulumiservice#379]
### What happened?

When deploying a new AgentPool using this provider, if the user runs `pulumi refresh`, then the resource looses the value for `agent_pool_id` and `token_value`.

Here are the steps to reproduce with the code below

1. Init a new stack
2. Select the new stack
3. Deploy the code below ✔
4. Run a `pulumi refresh` ✔
5. Run a `pulumi update` ✔
6. The 2  stacks output are gone, including in the stack state ❌

Upon inspection of the state after step 4 and step 5, the loss in values (`agent_pool_id` and `token_value`) seems to only occur once step 5 has completed (not at the end of step 4) (See state diff below).

I haven't been able to find a way to recover from this situation.

Note:
- I also noticed the field `description` changes to `""` after the `refresh` and back to `null` after the update.

### Example

```python
"""A Python Pulumi program"""

import pulumi
import pulumi_pulumiservice as pulumiservice

agent_pool = pulumiservice.AgentPool(
    name="agent-pool",
    resource_name="agent-pool",
    organization_name=pulumi.get_organization(),
)

pulumi.export("agent_pool_id", agent_pool.agent_pool_id)
pulumi.export("token_value", agent_pool.token_value)
```

### Output of `pulumi about`

```
CLI          
Version      3.129.0
Go Version   go1.22.6
Go Compiler  gc

Plugins
KIND      NAME           VERSION
resource  pulumiservice  0.23.1
language  python         unknown

Host     
OS       debian
Version  12.6
Arch     x86_64

This project is written in python: executable='/home/aureq/work/customers/zendesk/5662/venv/bin/python' version='3.11.9'

Current Stack: menfin/zendesk/5662

TYPE                            URN
pulumi:pulumi:Stack             urn:pulumi:5662::zendesk::pulumi:pulumi:Stack::zendesk-5662
pulumi:providers:pulumiservice  urn:pulumi:5662::zendesk::pulumi:providers:pulumiservice::default_0_23_1
pulumiservice:index:AgentPool   urn:pulumi:5662::zendesk::pulumiservice:index:AgentPool::agent-pool


Found no pending operations associated with 5662

Backend        
Name           pulumi.com
URL            https://app.pulumi.com/aureq
User           aureq
Organizations  aureq, team-ce, menfin, menfin-team, demo
Token type     personal

Dependencies:
NAME                  VERSION
pip                   24.2
pulumi-pulumiservice  0.23.1
setuptools            72.2.0
wheel                 0.44.0

Pulumi locates its logs in /tmp by default
```

### Additional context

```diff
--- out-2.json  2024-08-14 17:53:35.020438050 +1000
+++ out-3.json  2024-08-14 17:53:46.544210899 +1000
@@ -2,7 +2,7 @@
     "version": 3,
     "deployment": {
         "manifest": {
-            "time": "2024-08-14T17:53:23.170580504+10:00",
+            "time": "2024-08-14T17:53:41.904993372+10:00",
             "magic": "754c2596e98a48b601c099e42087c3a8042e7089feb1db31fe3368617a6dfdcc",
             "version": "v3.129.0"
         },
@@ -20,13 +20,6 @@
                 "urn": "urn:pulumi:5662::zendesk::pulumi:pulumi:Stack::zendesk-5662",
                 "custom": false,
                 "type": "pulumi:pulumi:Stack",
-                "outputs": {
-                    "agent_pool_id": "169029e7-c003-4a02-97cd-abcd70fd908a",
-                    "token_value": {
-                        "4dabf18193072939515e22adb298388d": "1b47061264138c4ac30d75fd1eb44270",
-                        "ciphertext": "AAABABwUzrBxJcN2UAzHYmGpuQwMpMbDHv8YUrKe34c9MzZ2sRRnCTPmpkc9vg2sTvNZ0F48UCpVAHcsgd4qPyWhRetMPLLFXEXDZ4vD"
-                    }
-                },
                 "created": "2024-08-14T07:53:03.11286988Z",
                 "modified": "2024-08-14T07:53:03.11286988Z",
                 "sourcePosition": "project:///venv/lib/python3.11/site-packages/pulumi/runtime/stack.py#139"
@@ -52,19 +45,13 @@
                 "id": "menfin/agent-pool/169029e7-c003-4a02-97cd-abcd70fd908a",
                 "type": "pulumiservice:index:AgentPool",
                 "inputs": {
-                    "description": "",
                     "name": "agent-pool",
                     "organizationName": "menfin"
                 },
                 "outputs": {
-                    "agentPoolId": "169029e7-c003-4a02-97cd-abcd70fd908a",
-                    "description": "",
+                    "description": null,
                     "name": "agent-pool",
-                    "organizationName": "menfin",
-                    "tokenValue": {
-                        "4dabf18193072939515e22adb298388d": "1b47061264138c4ac30d75fd1eb44270",
-                        "ciphertext": "AAABANcmoK/IjC+umKRb/p0RWtAT5x4qEeo7txTvZ82vPQ=="
-                    }
+                    "organizationName": "menfin"
                 },
                 "parent": "urn:pulumi:5662::zendesk::pulumi:pulumi:Stack::zendesk-5662",
                 "provider": "urn:pulumi:5662::zendesk::pulumi:providers:pulumiservice::default_0_23_1::5eb2a336-20fb-4e09-a7db-f09643687b6f",
@@ -76,7 +63,7 @@
                     "tokenValue"
                 ],
                 "created": "2024-08-14T07:53:05.2194437Z",
-                "modified": "2024-08-14T07:53:22.917751268Z",
+                "modified": "2024-08-14T07:53:41.520329129Z",
                 "sourcePosition": "project:///venv/lib/python3.11/site-packages/pulumi_pulumiservice/agent_pool.py#105"
             }
         ]
```

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by komalali** [comment]
@aureq I believe I've addressed the bulk of the issue but I was not able to repro the disappearing agent pool ID during my testing and it doesn't show up in the automated tests. If you have a clear example of how to repro that piece please open a new issue.

**Comment by aureq** [comment]
I confirm this is all working fine now. Thank you very much 🧡
